### DISCUSSION TITLE
Hardware design files

### DISCUSSION THREAD

**Discussion by Xiretza** [blackmagic-debug/blackmagic#1188] [Documentation]
https://github.com/blackmagic-debug/blackmagic-hardware has some design files for the BMP Mini and presumably the original BMP 1.0, but hasn't been updated since 2013. Is there any intention of releasing the newer BMPs as Open-Source Hardware designs as well? If not, is it possible to at least add a PDF schematic and mechanical drawing for BMP23 to https://black-magic.org/hardware.html#native-hardware? Unless I'm missing something, there is currently no documentation on how to use the new BMP23 features such as the unified debug connector or the better mechanical layout.

**Comment by esden** [comment]
Hi @Xiretza,

We are working on the documentation. That will be published soon enough. I have added the BMDU connector pinout documentation to the website a few days ago. https://black-magic.org/knowledge/pinouts.html#black-magic-debug-unified-connector-bmdu

More detailed documentation for the BMP V2.3 will come, including all the step by steps and what jumpers to change. As well as a mechanical drawing of the board. I have already shared a 3D printed case design with the people that asked for it on Discord.

As for the hardware design files, we do not publish those. The reason is that the hardware is what finances the development of BMD. All the current flurry of activity in the BMD repository and my efforts in documentation is 100% financed by the BMP hardware sales. None of it would be happening if people would stop buying the BMP hardware from us. Unfortunately if we publish the design files there are too many lazy opportunists that just want to make a buck on other's hard work. If you can't afford the BMP hardware the BMD firmware can be flashed on plenty of the ST-Link clones or BlackPill boards out there. We do put effort in keeping those supported. But the official BMP (aka. native) hardware will always be the best tested most up to date and recommended hardware to use.

That said, we do very much understand that the schematic is something of great value to those that have and use the BMP, and I am happy to send it to anyone who bought a BMP and asks me politely to send them the schematic.

Feel free to DM me on discord or send an email to info@1bitsquared.com with proof of purchase.

Cheers,
Piotr

**Reply by Xiretza** [reply]
That is an amazing overview of the common debug connectors, thanks a lot for creating those graphics! Much better than hunting around the net for dodgy PDFs every time.

Not publishing BMP PCB files makes sense, as you said there's really no reason for anyone to want to build this specific hardware themselves; BMF supports plenty of other targets.

I might get back to you on the schematic if the need arises, thanks for the offer!
